pkg/cmd/experimental/ipfailover/ipfailover.go

```diff
@@ -0,0 +1,173 @@
+package ipfailover
+
+import (
+	"fmt"
+	"io"
+
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/cmd/util/variable"
+	ipf "github.com/openshift/origin/pkg/ipfailover"
+	"github.com/openshift/origin/pkg/ipfailover/keepalived"
+)
+
+const shortDesc = "Configure or view IP Failover configuration"
+const description = `
+Configure or view IP Failover configuration
+
+This command helps to setup IP Failover configuration for an OpenShift
+environment. An administrator can configure IP failover on an entire
+cluster or as would normally be the case on a subset of nodes (as defined
+via a labelled selector).
+
+If an IP failover configuration does not exist with the given name,
+the --create flag can be passed to create a deployment configuration and
+service that will provide IP failover capability. If you are running in
+production, it is recommended that the labelled selector for the nodes
+matches atleast 2 nodes to ensure you have failover protection and that
+you provide a --replicas=<n> value that matches the number of nodes for
+the given labelled selector.
+
+
+Examples:
+  Check the default IP failover configuration ("ipfailover"):
+
+  $ %[1]s %[2]s
+
+  See what the IP failover configuration would look like if it is created:
+
+  $ %[1]s %[2]s -o json
+
+  Create an IP failover configuration if it does not already exist:
+
+  $ %[1]s %[2]s ipf --virtual-ips="10.1.1.1-4" --create
+
+  Create an IP failover configuration on a selection of nodes labelled
+  "router=us-west-ha" (on 4 nodes with 7 virtual IPs monitoring a service
+  listening on port 80 (aka the OpenShift router process).
+
+  $ %[1]s %[2]s ipfailover --selector="router=us-west-ha" --virtual-ips="1.2.3.4,10.1.1.100-104,5.6.7.8" --watch-port=80 --replicas=4 --create
+
+  Use a different IP failover config image and see the configuration:
+
+  $ %[1]s %[2]s ipf-alt --selector="jack=the-vipper" --virtual-ips="1.2.3.4" -o yaml --images=myrepo/myipfailover:mytag
+
+ALPHA: This command is currently being actively developed. It is intended
+       to simplify the administrative tasks of setting up a highly
+       available failover configuration.
+`
+
+func NewCmdIPFailoverConfig(f *clientcmd.Factory, parentName, name string, out io.Writer) *cobra.Command {
+	options := &ipf.IPFailoverConfigCmdOptions{
+		ImageTemplate:    variable.NewDefaultImageTemplate(),
+		Selector:         ipf.DefaultSelector,
+		ServicePort:      ipf.DefaultServicePort,
+		WatchPort:        ipf.DefaultWatchPort,
+		NetworkInterface: ipf.DefaultInterface,
+		Replicas:         1,
+	}
+
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("%s [<name>]", name),
+		Short: shortDesc,
+		Long:  fmt.Sprintf(description, parentName, name),
+		Run: func(cmd *cobra.Command, args []string) {
```

Comment:
This command needs to follow the pattern for the rest of our CLI commands - a RunProcess method that returns an error with cmdutil.CheckErr(err).  Also, you should always be returning errors from functions and glog fatal is reserved for a very few specific locations.